### ISSUE TITLE
withParam value could not be parsed as a JSON list

### ISSUE THREAD

**Issue by skoulouzis** [QCDIS/NaaVRE#42]
**Describe the bug**
Executing the workflow from laserfarm the Fetch Laz Files-1 fails with: 
'withParam value could not be parsed as a JSON list: {"laz_files": ["C_18HZ2.LAZ", "C_19HZ2.LAZ", "C_01GN2.LAZ", "C_50GZ2.LAZ"]}: json: cannot unmarshal object into Go value of type []v1alpha1.Item'

**Comment by skoulouzis** [comment]
Looks like is an error with how we output the list. 
Hardcoding as a list not as a dict works:
```yaml
metadata:
  name: workflow-test-sr7wq
  generateName: workflow-test-
  namespace: default
  uid: 08583273-49fe-4413-b699-183d325b8b59
  resourceVersion: '32025562'
  generation: 3
  creationTimestamp: '2022-01-10T11:44:14Z'
  labels:
    workflows.argoproj.io/creator: system-serviceaccount-default-jenkins
  managedFields:
    - manager: argo
      operation: Update
      apiVersion: argoproj.io/v1alpha1
      time: '2022-01-10T11:44:14Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:generateName': {}
          'f:labels':
            .: {}
            'f:workflows.argoproj.io/creator': {}
        'f:spec': {}
spec:
  templates:
    - name: workflow-test
      inputs: {}
      outputs: {}
      metadata: {}
      dag:
        tasks:
          - name: fetch-laz-files-1
            template: fetch-laz-files-1-tmp
            arguments: {}
          - name: retiling-1
            template: retiling-1-tmp
            arguments:
              parameters:
                - name: laz_files
                  value: '{{item}}'
                - name: param_login
                  value: '{{workflow.parameters.param_login}}'
                - name: param_password
                  value: '{{workflow.parameters.param_password}}'
                - name: param_hostname
                  value: '{{workflow.parameters.param_hostname}}'
            dependencies:
              - fetch-laz-files-1
            withParam: '{{tasks.fetch-laz-files-1.outputs.result}}'
    - name: fetch-laz-files-1-tmp
      inputs: {}
      outputs: {}
      metadata: {}
      container:
        name: ''
        image: 'alpine:latest'
        command:
          - sh
          - '-c'
        args:
          - |
            echo '["C_18HZ2.LAZ", "C_19HZ2.LAZ", "C_01GN2.LAZ", "C_50GZ2.LAZ"]'
        resources: {}
    - name: retiling-1-tmp
      inputs:
        parameters:
          - name: laz_files
          - name: param_login
          - name: param_password
          - name: param_hostname
      outputs: {}
      metadata: {}
      container:
        name: ''
        image: qcdis/retiling-1
        command:
          - /bin/bash
          - '-c'
        args:
          - >-
            source /venv/bin/activate; python retiling-1.py --laz_files
            {{inputs.parameters.laz_files}} --param_login
            {{workflow.parameters.param_login}} --param_password
            {{workflow.parameters.param_password}} --param_hostname
            {{workflow.parameters.param_hostname}};
        resources: {}
  entrypoint: workflow-test
  arguments:
    parameters:
      - name: param_login
        value: ''
      - name: param_password
        value: ''
      - name: param_hostname
        value: ''
```

**Comment by r-whites** [comment]
If the single output type, in case of parallelisation, is not 'list', the templating system falls back to the default case which incapsulates the stdout in a JSON. This cause Argo to not recognise the 'withParam' pattern since it expects a flat list instead of a dictionary.

**Comment by r-whites** [comment]
Fixed in: 09993c814b071f3ed6397dd6305f9c39f80b78e5
